qiime2/sdk/plugin_manager.py

```diff
@@ -93,18 +93,38 @@ def _integrate_plugin(self, plugin):
     # databases rollback/commit
 
     @property
+    def importable_formats(self):
+        """Return set of formats that are importable.
+
+        A format is importable in a QIIME 2 deployment if it can be transformed
+        into at least one of the canonical semantic type formats.
+
+        """
+        _importable_formats = set()
+        for record in self.formats.values():
+            for type_format in self.type_formats:
+                from_type = transform.ModelType.from_view_type(
+                    record.format)
+                to_type = transform.ModelType.from_view_type(
+                    type_format.format)
+                if from_type.has_transformation(to_type):
+                    _importable_formats.add(record)
+                    break
+        return _importable_formats
+
+    @property
     def importable_types(self):
         """Return set of concrete semantic types that are importable.
 
         A concrete semantic type is importable if it has an associated
         directory format.
 
         """
-        types = set()
-        for record in self.type_formats:
-            for type in record.type_expression:
-                types.add(type)
-        return types
+        _importable_types = set()
```

Comment:
Just a stylistic note that it's not necessary to prefix local variables with an underscore since they're local (*private*) to the function.